version: "3.8"

x-extra_hosts:
  &default-extra_hosts
  - "db_mssql: $DB_MSSQL"
  - "app_server: $APP_SERVER"
  - "{{ frontend.dns.api }}: $API_GATEWAY"
  - "{{ frontend.dns.frontend }}: $PORTAL_GATEWAY"
  - "{{ frontend.dns.cdn }}: $CDN_GATEWAY"

networks:
  servicos:
    external:
      name: servicos

services:
  PES00:
    image: "korp/viasoft.erp.person-frontend:3.2.x"
    container_name: "PES00"
    restart: on-failure:10
    extra_hosts: *default-extra_hosts
    environment:
      - ON_PREMISE_MODE=true
      - GATEWAY_URL={{ frontend.endpoints.gateway_url }}
      - CDN_URL={{ frontend.endpoints.cdn_url }}
      - FRONTEND_URL={{ frontend.endpoints.frontend_url }}
    networks:
      - servicos
    volumes:
      - "{{ self_signed_certs_directory }}/:{{ self_signed_certs_directory }}/"
  
  viasoft-erp-person:
    image: "korp/viasoft.erp.person:3.0.x"
    container_name: "Viasoft.ERP.Person"
    restart: on-failure:10
    extra_hosts: *default-extra_hosts
    environment:
      - ON_PREMISE_MODE=true
      - URL_CONSUL=http://consul-server:8500
    networks:
      - servicos
    volumes:
      - "{{ self_signed_certs_directory }}/:{{ self_signed_certs_directory }}/"
